core/src/main/scala/kafka/admin/AdminUtils.scala

```diff
@@ -280,30 +280,48 @@ object AdminUtils extends Logging with AdminUtilities {
           s"Assignment: $existingAssignment"))
 
     val partitionsToAdd = numPartitions - existingAssignment.size
-    if (partitionsToAdd <= 0)
+    if (partitionsToAdd < 0)
       throw new InvalidPartitionsException(
         s"The number of partitions for a topic can only be increased. " +
           s"Topic $topic currently has ${existingAssignment.size} partitions, " +
           s"$numPartitions would not be an increase.")
+    else if (partitionsToAdd == 0) {
+      if (replicaAssignment.isDefined) {
+        // Check the provided assignments are compatible with the tail of the existing
+        val added = replicaAssignment.get
+        // keyed according to the old assignment
+        val assignment2 = added.toList.sortBy { case (p, b) => p }.map { case (p, b) => p - existingAssignment.size + 1 -> b }.toMap
+        val blob = existingAssignment.toList.sortBy { case (p, b) => p }
+        val tailOld = existingAssignment.toList.sortBy { case (p, b) => p }.takeRight(added.size).toMap
+        if (tailOld != assignment2) {
```

Comment:
Does the != carry out desired comparison ?

Reply:
Yes,  it's comparing two `Map[Int, Seq[Int]]`

But that code still had a couple of debug variables in it, so I pushed a new version.